src/chronify/time_series_mapper_datetime.py

```diff
@@ -80,29 +89,46 @@ def _create_mapping(self) -> tuple[pd.DataFrame, MappingTableSchema]:
         """
         from_time_col = "from_" + self._from_time_config.time_column
         to_time_col = self._to_time_config.time_column
-        to_time_data = make_time_range_generator(self._to_time_config).list_timestamps()
+        from_time_data = make_time_range_generator(self._from_time_config).list_timestamps()
+        to_time_data = make_time_range_generator(
+            self._to_time_config, leap_day_adjustment=self._data_adjustment.leap_day_adjustment
+        ).list_timestamps()
+
+        dfs_from = pd.Series(from_time_data)
+        # If from_tz or to_tz is naive, use tz_localize TODO: separate these into classes
+        fm_tz = self._from_time_config.start.tzinfo
+        to_tz = self._to_time_config.start.tzinfo
+        match (fm_tz is None, to_tz is None):
+            case (True, False):
+                # get standard time zone of to_tz
+                to_tz_std = self._to_time_config.start.tzname()
+                # tz-naive time does not have skips/dups, so always localize in std tz first
+                dfs_from = dfs_from.dt.tz_localize(to_tz_std).dt.tz_convert(to_tz)
+            case (False, True):
+                dfs_from = dfs_from.dt.tz_localize(to_tz)
```

Comment:
A default case with a `raise Exception` would be good. Then the reader doesn't have to wonder about the other two cases.